### PR TITLE
added bugsnag destination

### PR DESCRIPTION
**Pull request by ruchiramoitra**
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/114)

### CODE REVIEW

File: integrations/Bugsnag/browser.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    );
+
+     this.setIntervalHandler = setInterval(
+      this.initBugsnagClient.bind(this),
+      1000
+    );
+  }
+
+   initBugsnagClient() {
+    if (window.bugsnag !== undefined && window.bugsnag !== void 0) {
```
Comment (sayan-mitra) [inline integrations/Bugsnag/browser.js:30]: what does void 0 does?
can we use a corresponding easy readable syntax

File: integrations/index.js [hunk 2]
```diff
@@ -35,6 +36,7 @@ const integrations = {
   FACEBOOK_PIXEL: FBPixel.default,
   LOTAME: Lotame.default,
   OPTIMIZELY: Optimizely.default,
+  BUGSNAG: Bugsnag.default
```
Comment (sayan-mitra) [inline integrations/index.js:39]: also add a corresponding entry in cname file. 
can you also check after making the change that selective destination for sending events is working fine for bugsnag

File: integrations/Bugsnag/browser.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  }
+
+   identify(rudderElement) {
+    let traits = rudderElement.message.context.traits;
+    let traitsFinal = {
+      id: rudderElement.message.userId,
+      name: traits.name,
+      email: traits.email
+    };
+    console.log(traitsFinal);
```
Comment (sayan-mitra) [inline integrations/Bugsnag/browser.js:54]: change to logger.debug

File: integrations/Bugsnag/browser.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+   isReady() {
+    logger.debug("in bugsnag isReady");
+    return !!window.bugsnagClient;
+  }
+
+   identify(rudderElement) {
+    let traits = rudderElement.message.context.traits;
+    let traitsFinal = {
+      id: rudderElement.message.userId,
```
Comment (sayan-mitra) [inline integrations/Bugsnag/browser.js:50]: should we do rudderElement.message.userId || rudderElement.message.anonymousId ?

File: integrations/Bugsnag/browser.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    logger.debug("in bugsnag isLoaded");
+    return !!window.bugsnagClient;
+  }
+
+   isReady() {
+    logger.debug("in bugsnag isReady");
+    return !!window.bugsnagClient;
+  }
+
+   identify(rudderElement) {
```
Comment (sayan-mitra) [inline integrations/Bugsnag/browser.js:47]: also check for page/track etc api calls, the sdk is not crashing.
Ideally it should be logging the error in console and not crash, still check once

File: integrations/Bugsnag/browser.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
+import logger from "../../utils/logUtil";
+import  ScriptLoader  from "../ScriptLoader";
+import {
```
Comment (sayan-mitra) [inline integrations/Bugsnag/browser.js:3]: please remove unused imports

File: integrations/Bugsnag/browser.js (hunk 1 above)
Comment (ruchiramoitra) [reply integrations/Bugsnag/browser.js:30]: Similarly it was done when initialising other destinations. To check if undefined or not.. Should I remove it ?

File: integrations/Bugsnag/browser.js (hunk 4 above)
Comment (ruchiramoitra) [reply integrations/Bugsnag/browser.js:50]: yes. added it.

File: integrations/Bugsnag/browser.js (hunk 5 above)
Comment (ruchiramoitra) [reply integrations/Bugsnag/browser.js:47]: no called page and track the sdk is not crashing

File: integrations/Bugsnag/browser.js (hunk 6 above)
Comment (ruchiramoitra) [reply integrations/Bugsnag/browser.js:3]: removed

File: integrations/Bugsnag/browser.js (hunk 3 above)
Comment (ruchiramoitra) [reply integrations/Bugsnag/browser.js:54]: changed

**Comment by sayan-mitra** [comment]
👍

**Pull request merged**